
/// Catches any throws and switches if to af failure after printing the error.
public func printError(_ error: Error) {
    var faroError = error
    if !(error is FaroError) {
        faroError = FaroError.nonFaroError(error)
    }
    switch faroError as! FaroError {
    case .general:
        print("ðŸ’£ General service error")
    case .invalidUrl(let url):
        print("ðŸ’£invalid url: \(url)")
    case .invalidResponseData(_):
        print("ðŸ’£ Invalid response data")
    case .invalidAuthentication:
        print("ðŸ’£ Invalid authentication")
    case .shouldOverride:
        print("ðŸ’£ You should override this method")
    case .nonFaroError(let nonFaroError):
        print("ðŸ’£ Error from service: \(nonFaroError)")
    case .rootNodeNotFound(json: let json):
        print("ðŸ’£ Could not find root node in json: \(json)")
    case .networkError(let networkError):
        print("ðŸ’£ HTTP error: \(networkError)")
    case .emptyCollection:
        print("empty collection")
    case .emptyKey:
        print("missing key")
    case .emptyValue(let key):
        print("no value for key " + key)
    case .malformed(let info):
        print(info)
    case .serializationError:
        print("serialization error")
    }
    
}
